sandy/core/cov.py

```diff
@@ -393,20 +532,197 @@ def decompose(self):
+                           index=var_.index, columns=var_.index)
+        return cls(cov)
+
+    @classmethod
+    def from_stdev(cls, std):
+        """
+        Construct the covariance matrix from the standard deviation vector.
+
+        Parameters
+        ----------
+        std : `pandas.Series`
+            Standard deviations vector.
+
+        Returns
+        -------
+        `CategoryCov`
+            Object containing the covariance matrix.
+
+        Example
+        -------
+        >>> S = pd.Series(np.array([0, 2, 3]), index=pd.Index([1, 2, 3]))
+        >>> cov = sandy.CategoryCov.from_stdev(S)
+        >>> cov
+                    1           2           3
+        1 0.00000e+00 0.00000e+00 0.00000e+00
+        2 0.00000e+00 4.00000e+00 0.00000e+00
+        3 0.00000e+00 0.00000e+00 9.00000e+00
+
+        >>> assert type(cov) is sandy.CategoryCov
+
+        >>> S = sandy.CategoryCov.from_stdev((1, 2, 3))
+        >>> S
+                    0           1           2
+        0 1.00000e+00 0.00000e+00 0.00000e+00
+        1 0.00000e+00 4.00000e+00 0.00000e+00
+        2 0.00000e+00 0.00000e+00 9.00000e+00
+
+        >>> assert type(S) is sandy.CategoryCov
+        >>> assert type(sandy.CategoryCov.from_stdev([1, 2, 3])) is sandy.CategoryCov
+        """
+        std_ = pd.Series(std)
+        std_ = std_ * std_
+        return CategoryCov.from_var(std_)
+
+    def _gls_sensitivity(self, S, Vy, threshold=None):
+        """
+        Method to obtain sensitivity according to GLS
+
+        Parameters
+        ----------
+        S : 2D iterable
+            Sensitivity square matrix (MXM).
+        Vy : 1D iterable
+            Extra Covariance vector (MX1).
+        threshold : `int`, optional
+            threshold to avoid numerical fluctuations. The default is None.
+
+        Returns
+        -------
+        `CategoryCov`
+            DESCRIPTION.
+
+        """
+        S_ = pd.DataFrame(S)
+        Vy_ = pd.Series(Vy)
+        # GLS_sensitivity:
         Vx = self.data.values
-        V = S.T.dot(Vx.dot(S)) + Vy
-        C = Vx - Vx.dot(S.dot(np.linalg.inv(V).dot(S.T.dot(Vx))))
-        return self.__class__(C)
+        M = S_.T.dot(Vx.dot(S_)) + Vy_
+        M_inv = sandy.CategoryCov(M).invert()
+        sensitivity = Vx.dot(S_).dot(M_inv).dot(S_.T)
+        if threshold is not None:
+            sensitivity[sensitivity < threshold] = 0
+        return self.__class__(sensitivity)
+
+    def gls_update(self, Vy, S, delta, threshold=None):
+        """
+        
+
+        Parameters
+        ----------
+        Vy : 1D iterable
+            Extra Covariance vector (MX1).
+        S : 2D iterable
+            Sensitivity square matrix (MXM).
+        delta : 1D iterable
+            Perturbed vector minus non perturbed vector.
+        threshold : `int`, optional
+            Thereshold to avoid numerical fluctuations. The default is None.
+
+        Returns
+        -------
+        TYPE
+            DESCRIPTION.
+

```

Comment:
can we define a very simple test...maybe a 2x2 matrix for the prior, a 2x1 for the sensitivity and a 1x1 for the new info